httpService/pkg/user/transport.go

```diff
@@ -0,0 +1,81 @@
+package user
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/timoteoBone/project-microservice/grpcService/pkg/entities"
+	"github.com/timoteoBone/project-microservice/grpcService/pkg/errors"
+	myerr "github.com/timoteoBone/project-microservice/grpcService/pkg/errors"
+
+	"github.com/go-kit/kit/log"
+	httptransport "github.com/go-kit/kit/transport/http"
+)
+
+func NewHTTPSrv(endpoint Endpoints, logger log.Logger) http.Handler {
+	rt := mux.NewRouter()
+
+	options := []httptransport.ServerOption{
+		httptransport.ServerErrorLogger(logger),
+		httptransport.ServerErrorEncoder(encodeErrorResponse),
+	}
+
+	rt.Methods("POST").Path("/user").Handler(httptransport.NewServer(
+		endpoint.CreateUs,
+		decodeCreateUserReq,
+		encodeCreateUserResp,
+		options...,
+	))
+
+	rt.Methods("GET").Path("/user/{id}").Handler(httptransport.NewServer(
+		endpoint.GetUs,
+		decodeGetUserReq,
+		encodeGetUserResp,
+		options...,
+	))
+
+	return rt
+}
+
+func decodeCreateUserReq(ctx context.Context, r *http.Request) (interface{}, error) {
```

Comment:
same in the grpc, separate decodes and encoders, and separate from transport file